### PR TITLE
GEOT-5296: Apply WFS timeout to read timeout

### PR DESCRIPTION
**Pull request by awaterme**
WFSConfig.timeoutMillis is applied to read timeout of HTTPClient, too.
See [GEOT-5296](https://osgeo-org.atlassian.net/browse/GEOT-5296)

### CODE REVIEW

File: modules/unsupported/wfs-ng/src/main/java/org/geotools/data/wfs/WFSDataStoreFactory.java [hunk 1]
```diff
@@ -107,6 +107,7 @@ public WFSDataStore createDataStore(final Map<String, Serializable> params)
         http.setPassword(config.getPassword());
         int timeoutMillis = config.getTimeoutMillis();
         http.setConnectTimeout(timeoutMillis / 1000);
+        http.setReadTimeout(timeoutMillis / 1000);
```
Comment (aaime) [inline modules/unsupported/wfs-ng/src/main/java/org/geotools/data/wfs/WFSDataStoreFactory.java]: Believe the very same change should be applied to WFSDataAccessFactory

**Comment by awaterme** [comment]
Adjusted WFSDataAccessFactory, too.

**Comment by aaime** [comment]
I see we have to update the copyright header year, but besides that, it looks good to me.
@NielsCharlier is the author of this module, what do you think Niels?

**Comment by NielsCharlier** [comment]
Yeah that is fine (not the initial creator of the module, but indeed did a lot of work to make it usable)

**Pull request merged**